exec/java-exec/src/main/java/org/apache/drill/exec/ExecConstants.java

```diff
@@ -1166,4 +1166,8 @@ public static String bootDefaultFor(String name) {
 
   // Extra private plugin classes, used for testing
   public static final String PRIVATE_CONNECTORS = "drill.exec.storage.private_connectors";
+
+  public static final String ENABLE_DYNAMIC_CREDIT_BASED_FC = "exec.enable_dynamic_fc";
+  public static final BooleanValidator ENABLE_DYNAMIC_CREDIT_BASED_FC_VALIDATOR = new BooleanValidator(
+          ENABLE_DYNAMIC_CREDIT_BASED_FC, new OptionDescription("Enable dynamic credit based flow control."));
```

Comment:
Please add more details for user can understand the benefits of dynamic credit flow, for example `... which allows ...` and `... should be used when ...`.